apps/project/serializers.py

```diff
@@ -446,10 +451,17 @@ def create(self, validated_data):
 
 class ProjectUserGroupSerializer(serializers.ModelSerializer):
     title = serializers.CharField(source='usergroup.title', read_only=True)
+    role_title = serializers.CharField(source='role.title', read_only=True)
```

Comment:
Let's define SimpleRoleSerializer and use that instead.

role_details = SimpleRoleSerializer(source='role', read_only=True)

Make sure to also use select related. So role_id and role_title should be enough.